suite/tests/linux/sigcontext.c

```diff
@@ -323,8 +325,21 @@ main(int argc, char *argv[])
         MOVE_TO_YMM(buf, 15)
 #        endif
 #    endif
-        /* now make sure they show up in signal context */
-        kill(getpid(), SIGUSR2);
+        /* Now make sure they show up in the signal context.
+         * But, it's not safe to make a regular call, as the compiler will
+         * insert VZEROUPPER for some configurations.
```

Comment:
Do we know why the compiler does that?

Reply:
Yes, it is recommended by Intel to avoid a performance penalty when switching between upper-zeroing VEX-encoded XMM operations and legacy upper-preserving SSE* operations: so doing it at transitions to code you don't control makes sense.